pkg/puller/puller.go

```diff
@@ -354,87 +257,79 @@ func (p *Puller) histSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 			return
 		}
+		if syncErr != nil {
+			p.metrics.HistWorkerErrCounter.Inc()
+			p.logger.Error(syncErr, "histSyncWorker syncing interval failed", "peer_address", peer, "bin", bin, "cursor", cur, "start", s, "topmost", top)
+		}
+
 		loggerV2.Debug("histSyncWorker pulled", "bin", bin, "start", s, "topmost", top, "peer_address", peer)
 	}
 }
 
 func (p *Puller) liveSyncWorker(ctx context.Context, peer swarm.Address, bin uint8, cur uint64) {
-	loggerV2 := p.logger.V(2).Register()
+	loggerV2 := p.logger
 
 	defer p.wg.Done()
-	loggerV2.Debug("liveSyncWorker starting", "peer_address", peer, "bin", bin, "cursor", cur)
+	p.logger.Debug("liveSyncWorker starting", "peer_address", peer, "bin", bin, "cursor", cur)
 	from := cur + 1
+
+	sleep := false
+
 	for {
 		p.metrics.LiveWorkerIterCounter.Inc()
+
+		if sleep {
+			select {
+			case <-ctx.Done():
+				loggerV2.Debug("liveSyncWorker context cancelled", "peer_address", peer, "bin", bin, "cursor", cur)
+				return
+			case <-time.After(syncSleepDur):
+			}
+			sleep = false
+		}
+
 		select {
-		case <-p.quit:
-			loggerV2.Debug("liveSyncWorker quit on shutdown", "peer_address", peer, "bin", bin, "cursor", cur)
-			return
 		case <-ctx.Done():
 			loggerV2.Debug("liveSyncWorker context cancelled", "peer_address", peer, "bin", bin, "cursor", cur)
 			return
 		default:
 		}
-		top, ruid, err := p.syncer.SyncInterval(ctx, peer, bin, from, pullsync.MaxCursor)
-		if err != nil {
-			loggerV2.Debug("liveSyncWorker exit on sync error", "peer_address", peer, "bin", bin, "from", from, "error", err)
-			if ruid == 0 {
-				p.metrics.LiveWorkerErrCounter.Inc()
-			}
 
-			// since we use bin context cancellation to cancel interval
-			// sync operations, the context can be expired here, causing us
-			// to try to send a message with an expired context, which is
-			// bound to fail.
-			ctxC, cancelC := context.WithTimeout(context.Background(), 10*time.Second)
-			defer cancelC()
-			if err := p.syncer.CancelRuid(ctxC, peer, ruid); err != nil {
-				loggerV2.Debug("histSyncWorker cancel ruid failed", "error", err)
-			}
-			return
-		}
+		top, _, syncErr := p.syncer.SyncInterval(ctx, peer, bin, from, pullsync.MaxCursor)
 		if top == math.MaxUint64 {
 			p.metrics.MaxUintErrCounter.Inc()
 			return
 		}
-		err = p.addPeerInterval(peer, bin, from, top)
+		err := p.addPeerInterval(peer, bin, from, top)
 		if err != nil {
 			p.metrics.LiveWorkerErrCounter.Inc()
 			p.logger.Error(err, "liveSyncWorker exit on add peer interval", "peer_address", peer, "bin", bin, "from", from, "error", err)
 			return
 		}
-		loggerV2.Debug("liveSyncWorker pulled bin failed", "bin", bin, "from", from, "topmost", top, "peer_address", peer)
+
+		if syncErr != nil {
+			p.metrics.LiveWorkerErrCounter.Inc()
+			p.logger.Error(syncErr, "liveSyncWorker sync error", "peer_address", peer, "bin", bin, "from", from, "topmost", top)
+			sleep = true
+		}
```

Comment:
`syncErr` should be handled right after `p.syncer.SyncInterval` method is called; also name should be follow same convention.